File: in_Java/0905 Sort Array By Parity.java
class Solution {
    public int[] sortArrayByParity(int[] A) {
        int[] res = new int[A.length];
        int i = 0;
        int j = res.length-1;
        for(int x : A){
            if(x % 2 == 0){
                res[i++] = x;
            }else{
                res[j--] = x;
            }
        }
        return res;
    }
}

File: in_Python/0905 Sort Array By Parity.py
class Solution(object):
    def sortArrayByParity(self, A):
        """
        :type A: List[int]
        :rtype: List[int]
        """
        res = []
        for i in range(len(A)):
            res.append(0)
        i = 0
        j = -1
        for a in A:
            if a % 2 == 0:
                res[i] = a
                i += 1
            else:
                res[j] = a
                j -= 1
        return res
                